cmd/cliclient/cleanup.go

```diff
@@ -0,0 +1,61 @@
+package cliclient
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ory/x/configx"
+
+	"github.com/spf13/cobra"
+
+	"github.com/ory/kratos/driver"
+	"github.com/ory/kratos/driver/config"
+	"github.com/ory/x/cmdx"
+	"github.com/ory/x/flagx"
+)
+
+type CleanupHandler struct{}
+
+func NewCleanupHandler() *CleanupHandler {
+	return &CleanupHandler{}
+}
+
+func (h *CleanupHandler) CleanupSQL(cmd *cobra.Command, args []string) {
+	var d driver.Registry
+
+	if flagx.MustGetBool(cmd, "read-from-env") {
+		d = driver.NewWithoutInit(
+			cmd.Context(),
+			cmd.ErrOrStderr(),
+			configx.WithFlags(cmd.Flags()),
+			configx.SkipValidation())
+		if len(d.Config(cmd.Context()).DSN()) == 0 {
+			fmt.Println(cmd.UsageString())
+			fmt.Println("")
+			fmt.Println("When using flag -e, environment variable DSN must be set")
+			os.Exit(1)
```

Comment:
As tests will always fail if you call `os.Exit` we instead use `cobra.Command.RunE` and propagate errors up. As in this case you already printed the error message, you can just return `cmdx.FailSilently(cmd)`. It will silence the standard cobra output and results in an error exit code being set.

Reply:
I think I did what you had in mind